torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1579,7 +1594,10 @@ void ProcessGroupNCCL::watchdogHandler() {
               // Set shutdown mode, so the heartbeat monitor thread will not
               // abort process immediately.
               collectiveDebugInfoMode_.store(true);
-              std::vector<uint8_t> vec(1);
+              int rank = globalRank();
+              auto vec = std::vector<uint8_t>(
+                  reinterpret_cast<uint8_t*>(&rank),
+                  reinterpret_cast<uint8_t*>(&rank) + sizeof(int));
```

Comment:
ok im assuming set/get always operate on byte vectors.  if so, this looks correct but i'd prefer to use a more specific type instead of 'int'.  size_t? or what does globalRank() actually return?